src/main/java/calculator/Calculator.java

```diff
@@ -0,0 +1,102 @@
+package calculator;
+
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static calculator.Constant.*;
+
+public class Calculator {
+
+    private static final Pattern COMPILED_CUSTOM_SPLIT_REGEX =
+            Pattern.compile(CUSTOM_DELIMITER_PATTERN);
+
+    public int splitAndSum(String input) {
+        if (isInputNullOrEmpty(input)) {
+            return RESULT_OF_EMPTY_INPUT;
+        }
+        if (customSeparatorExists(input)) {
+            return calculateWithCustomDelimiter(input);
+        }
+        if (isInputOneNumberString(input)) {
+            return calculateOneNumberInputString(input);
+        }
+        return calculateWithDefaultSeparator(input);
+    }
+
+    private boolean isInputNullOrEmpty(String input) {
+        return input == null || input.isEmpty() || input.isBlank();
+    }
+
+    private boolean customSeparatorExists(String input) {
+        return input.matches(CUSTOM_DELIMITER_PATTERN);
+    }
+
+    private int calculateWithCustomDelimiter(String input) {
+        checkTokensAreInvalid(input);
+        String s = input.replaceAll(NON_DIGIT_CHARACTER_PATTERN, BLANK)
+                .trim();
+        String[] tokens = s.split(BLANK);
+        return calculateSum(tokens);
+    }
+
+    private void checkTokensAreInvalid(String input) {
+        try {
+            splitCustomInputAndParseToInt(input);
+        } catch (NumberFormatException e) {
+            throw new IllegalArgumentException("지정된 커스텀 구분자로 숫자를 구분해야 합니다.");
+        }
+    }
+
+    private void splitCustomInputAndParseToInt(String input) {
+        Matcher m = COMPILED_CUSTOM_SPLIT_REGEX.matcher(input);
+        if (m.find()) {
+            String customDelimiter = m.group(CUSTOM_DELIMITER_GROUP_INDEX);
+            String[] tokens = m.group(CUSTOM_INPUT_NUMBER_GROUP_INDEX)
+                    .split(customDelimiter);
+            Arrays.stream(tokens).forEach(token -> {
+                Integer.parseInt(token.trim());
+            });
+        }
+    }
+
+    private Integer calculateSum(String[] tokens) {
+        return Arrays.stream(tokens)
+                .map(s -> Integer.parseInt(s.trim()))
+                .reduce(INITIAL_VALUE_OF_SUM, (a, b) -> a + b);
```

Comment:
이 방식은 어떨까요? [모던 자바 인 액션 3장. 람다 표현식](https://www.skullkim-dev.com/2022/02/14/3%EC%9E%A5.-%EB%9E%8C%EB%8B%A4-%ED%91%9C%ED%98%84%EC%8B%9D/)을 참고해 주세요~
```suggestion
                .reduce(INITIAL_VALUE_OF_SUM, Integer::sum);
```

Reply:
이런식으로도 쓸 수 있군요. 알려주신 방식이 훨씬 간결하고 알기 쉬운 것 같아요.
올려주신 블로그 링크도 읽어볼게요.